Configuration/Skimming/src/LeptonSkimming.cc

```diff
@@ -0,0 +1,487 @@
+    }
+//we know that at least a trigger fired
+//find the total max
+  std::sort( max_per_trigger.begin(), max_per_trigger.end(),
+         [](const std::vector<float>& a, const std::vector<float>& b) {
+               return a[0] > b[0];
+                });
+  return  max_per_trigger.at(0); 
+
+}
+
+
+
+
+// ------------ method called on each new Event  ------------
+bool
+LeptonSkimming::filter(edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+ 
+  using namespace std;
+  using namespace edm;
+  using namespace reco;
+  using namespace trigger;
+  // using namespace PhysicsTools;
+
+  test_ev++;
+  //Get a few collections to apply basic electron ID
+  //Get electrons
+  edm::Handle<std::vector<reco::GsfElectron> > electrons;
+  iEvent.getByToken(electronsToken_, electrons);
+
+ 
+  edm::Handle<std::vector<reco::Muon>> muons;
+   iEvent.getByToken(muonsToken_,muons);
+
+  //Get conversions
+  edm::Handle<reco::ConversionCollection> conversions;
+  iEvent.getByToken(conversionsToken_, conversions);    
+  // Get the beam spot
+  edm::Handle<reco::BeamSpot> theBeamSpot;
+  iEvent.getByToken(beamSpotToken_,theBeamSpot);  
+  //Get vertices 
+  edm::Handle<std::vector<reco::Vertex>> vertices;
+  iEvent.getByToken(vtxToken_, vertices);
+  //continue if there are no vertices
+  if (vertices->empty()) return false;
+  edm::Handle<vector<reco::Track>> tracks;
+  iEvent.getByToken(Tracks_, tracks);
+  edm::Handle<edm::TriggerResults> trigResults;
+  iEvent.getByToken(trgresultsToken_, trigResults);
+  edm::ESHandle<MagneticField> bFieldHandle;
+  iSetup.get<IdealMagneticFieldRecord>().get(bFieldHandle);
+  KalmanVertexFitter theKalmanFitter(false);
+  TransientVertex LLvertex;
+  
+
+  vertex_x=-1000000;  vertex_y=-1000000;  vertex_z=-1000000;  
+  beam_x=-9999,beam_y=-9999,beam_z=-99999;
+ 
+  // trigger1=0; trigger2=0; trigger3=0; trigger4=0; trigger5=0; trigger6=0;
+  nmuons=0; nel=0; ntracks=0;
+    
+  
+  SelectedTrgObj_PtEtaPhiCharge.clear(); SelectedMu_index=-1;
+  SelectedMu_DR=1000; muon_pt.clear(); muon_eta.clear(); muon_phi.clear();
+  Result=false;  el_pt.clear(); el_eta.clear(); el_phi.clear();
+  Trk_container.clear();  MuTracks.clear();  ElTracks.clear(); 
+  object_container.clear(); object_id.clear();  cleanedTracks.clear(); 
+  Epair_ObjectId.clear(); muon_soft.clear(); muon_medium.clear(); muon_tight.clear();
+  Epair_ObjectIndex.clear(); cleanedObjTracks.clear(); cleanedPairTracks.clear();
+    Epair_ObjectIndex.clear(); Epair_ObjectId.clear(); Epair_TrkIndex.clear();
+  //internal stuff
+   ZvertexTrg=-100000000;
+
+   for (const reco::Vertex& vtx : *vertices) {
+    bool isFake = vtx.isFake();
+    if ( isFake) continue;
+        vertex_x=vtx.x();  vertex_y=vtx.y();  vertex_z=vtx.z(); 
+	if ( vertex_x!=-10000000) break;
+    }
+  if (vertex_x==-1000000)
+         return false;
+  reco::TrackBase::Point  vertex_point;
+  vertex_point.SetCoordinates(vertex_x,vertex_y,vertex_z);
```

Comment:
all of this copying of vertex location seems unneeded

Reply:
i only keep this for the correct dxy. We need the vertex_point. If at the end is not needed in the code i will remove it. In general i rewrote it with less copying